core_v14/src/value/serialize.rs

```diff
@@ -0,0 +1,189 @@
+// This file is part of substrate-desub.
+//
+// substrate-desub is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// substrate-desub is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with substrate-desub.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::{Composite, Primitive, Value, Variant};
+use serde::{
+	ser::{SerializeMap, SerializeSeq},
+	Serialize,
+};
+
+impl Serialize for Value {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		match self {
+			Value::Composite(val) => val.serialize(serializer),
+			Value::Variant(val) => val.serialize(serializer),
+			Value::BitSequence(val) => val.serialize(serializer),
+			Value::Primitive(val) => val.serialize(serializer),
+		}
+	}
+}
+
+impl Serialize for Composite {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		match self {
+			Composite::Named(vals) => {
+				let mut map = serializer.serialize_map(Some(vals.len()))?;
+				for (key, val) in vals {
+					map.serialize_entry(key, val)?;
+				}
+				map.end()
+			}
+			Composite::Unnamed(vals) => {
+				let mut seq = serializer.serialize_seq(Some(vals.len()))?;
+				for val in vals {
+					seq.serialize_element(val)?;
+				}
+				seq.end()
+			}
+		}
+	}
+}
+
+impl Serialize for Primitive {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		// Delegate to the serialization strategy used by the primitive types.
+		match self {
+			Primitive::Bool(v) => v.serialize(serializer),
+			Primitive::Char(v) => v.serialize(serializer),
+			Primitive::Str(v) => v.serialize(serializer),
+			Primitive::U8(v) => v.serialize(serializer),
+			Primitive::U16(v) => v.serialize(serializer),
+			Primitive::U32(v) => v.serialize(serializer),
+			Primitive::U64(v) => v.serialize(serializer),
+			Primitive::U128(v) => v.serialize(serializer),
+			Primitive::U256(v) => v.serialize(serializer),
+			Primitive::I8(v) => v.serialize(serializer),
+			Primitive::I16(v) => v.serialize(serializer),
+			Primitive::I32(v) => v.serialize(serializer),
+			Primitive::I64(v) => v.serialize(serializer),
+			Primitive::I128(v) => v.serialize(serializer),
+			Primitive::I256(v) => v.serialize(serializer),
+		}
+	}
+}
+
+impl Serialize for Variant {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		// W can't use the enum serializing in the serde data model because that requires static
```

Comment:
?

```suggestion
		// We can't use the enum serializing in the serde data model because that requires static
```

Reply:
I wish I could blame that on a british abbreviation :D